File: src/main/scala/com/bombproj/model/ProjectMember.java
package com.bombproj.model;

import com.queryflow.annotation.Column;
import com.queryflow.annotation.Table;
import lombok.Data;

import java.util.Date;

@Data
@Table("project_member")
public class ProjectMember {

    private String id;
    private Integer state;
    private Date createTime;
    private Date updateTime;
    private String projectId;
    private String userId;
    private String permission;

    @Column(exists = false)
    private String userEmail;
    @Column(exists = false)
    private String userPhone;
    @Column(exists = false)
    private String userName;
    @Column(exists = false)
    private String userNickName;

}

File: src/main/scala/com/bombproj/model/ProjectMember.scala
package com.bombproj.model

import java.util.Date

import com.queryflow.annotation.{Column, Table}

import scala.beans.BeanProperty

@SerialVersionUID(1L)
@Table("project_member")
case class ProjectMember() {

    @BeanProperty var id: String = _
    @BeanProperty var state: Int = _
    @BeanProperty var createTime: Date = _
    @BeanProperty var updateTime: Date = _
    @BeanProperty var projectId: String = _
    @BeanProperty var userId: String = _
    @BeanProperty var permission: String = _

    @BeanProperty
    @Column(exists = false) var userEmail: String = _
    @BeanProperty
    @Column(exists = false) var userPhone: String = _
    @BeanProperty
    @Column(exists = false) var userName: String = _
    @BeanProperty
    @Column(exists = false) var userNickName: String = _

}
